src/web/extractors.rs

```diff
@@ -284,14 +289,11 @@ impl FromRequest<ServerState> for BsoBody {
                 ));
             }
         }
-        let mut config = JsonConfig::default();
-        let max_request_size = req.state().limits.max_request_bytes as usize;
-        config
-            .limit(max_request_size)
-            .content_type(|ct| ct == mime::TEXT_PLAIN);
-
-        let max_payload_size = req.state().limits.max_record_payload_bytes as usize;
-        let fut = <Json<BsoBody>>::from_request(req, &config)
+        let state = req.app_data::<ServerState>().unwrap();
+
+        let max_payload_size = state.limits.max_record_payload_bytes as usize;
```

Comment:
Probably, still, since `from_request` is a magic method, we can't always assume that the only path to this method is from `build_app`. It may be safer to check again.